CompilerSource/JDI/src/Parser/readers/read_type.cpp

```diff
@@ -336,7 +336,7 @@ int jdip::context_parser::read_referencers(ref_stack &refs, const full_type& ft,
         if (!d) return 1;
         
         if (token.type == TT_MEMBEROF) {
-          if (!(d->flags & DEF_CLASS)) {
+          if (!(d->flags & DEF_CLASS) && !(d->flags & DEF_TYPENAME)) {
```

Comment:
This is somewhat dangerous; line 343 treats this as a `definition_class*`. That cast isn't valid if this original condition isn't true.

Reply:
so what's the safe way to handle this then? stl has template<typename _Ret, typename _Tp> but _Tp is a class not a type so I get ```Member pointer to non-class `_Tp'```

Reply:
I would have to dig into how that other method is using its operand to know for sure. I'm not sure what it needs from a class that it can't get from a regular template parameter. The simplest short-term solution would be to revert this line, and have template parameters declared as `typename` or `class` actually extend `definition_class`.